packages/runtime/container-runtime/src/test/gc/gcTelemetryTracker.spec.ts

```diff
@@ -0,0 +1,737 @@
+			// Mark nodes 2 and 3 as unreferenced.
+			markNodesUnreferenced([nodes[2], nodes[3]]);
+
+			// Advance the clock to trigger inactive timeout and validate that inactive events are as expected.
+			clock.tick(inactiveTimeoutMs + 1);
+			updateNodes(nodes);
+			await generateAllEvents(isSummarizerClient);
+			assertMatchEvents(
+				[
+					{
+						eventName: "GarbageCollector:InactiveObject_Changed",
+						timeout: inactiveTimeoutMs,
+						id: nodes[2],
+					},
+					{
+						eventName: "GarbageCollector:InactiveObject_Loaded",
+						timeout: inactiveTimeoutMs,
+						id: nodes[2],
+					},
+					{
+						eventName: "GarbageCollector:InactiveObject_Changed",
+						timeout: inactiveTimeoutMs,
+						id: nodes[3],
+					},
+					{
+						eventName: "GarbageCollector:InactiveObject_Loaded",
+						timeout: inactiveTimeoutMs,
+						id: nodes[3],
+					},
+				],
+				"inactive events not as expected",
+			);
+
+			// Advance the clock to trigger sweep timeout and validate that sweep ready events are as expected.
+			clock.tick(sweepTimeoutMs - inactiveTimeoutMs);
+			updateNodes(nodes);
+			await generateAllEvents(isSummarizerClient);
+			assertMatchEvents(
+				[
+					{
+						eventName: "GarbageCollector:SweepReadyObject_Changed",
+						timeout: sweepTimeoutMs,
+						id: nodes[2],
+					},
+					{
+						eventName: "GarbageCollector:SweepReadyObject_Loaded",
+						timeout: sweepTimeoutMs,
+						id: nodes[2],
+					},
+					{
+						eventName: "GarbageCollector:SweepReadyObject_Changed",
+						timeout: sweepTimeoutMs,
+						id: nodes[3],
+					},
+					{
+						eventName: "GarbageCollector:SweepReadyObject_Loaded",
+						timeout: sweepTimeoutMs,
+						id: nodes[3],
+					},
+				],
+				"sweep ready events not as expected",
+			);
+		});
+
+		it("generates tombstone revived events when nodes are used after they are tombstoned", async () => {
+			telemetryTracker = createTelemetryTracker(true /* enable Sweep */, isSummarizerClient);
+			// Mark node 2 as unreferenced.
+			markNodesUnreferenced([nodes[2]]);
+
+			// Advance the clock to trigger sweep timeout and validate that tombstone revived event is as expected.
+			clock.tick(sweepTimeoutMs + 1);
+			reviveNode(nodes[1], nodes[2], true /* isTombstoned */);
+			mockLogger.assertMatch(
+				[
+					{
+						eventName: "GarbageCollector:GC_Tombstone_DataStore_Revived",
+						url: nodes[2],
+					},
+				],
+				"inactive events not as expected",
+			);
+		});
+
+		/** Tests that validate either inactive or sweep events are logged as expected. */
+		const inactiveOrSweepEventTests = (
+			timeout: number,
+			mode: "inactive" | "sweep",
+			revivedEventName: string,
+			changedEventName: string,
+			loadedEventName: string,
+			expectDeleteLogs?: boolean,
+		) => {
+			const deleteEventName = "GarbageCollector:GCObjectDeleted";
+
+			// Validates that no unexpected event has been fired.
+			function validateNoUnexpectedEvents() {
```

Comment:
I'd rename to just `validateNoEvents` - It's saying _any_ event would be unexpected.  And would be clearer when reading the tests.